src/Psl/Vec/chunk_with_keys.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Vec;
+
+use Psl;
+
+/**
+ * The same as chunk(), but preserving keys.
+ *
+ * Examples:
+ *
+ *     Vec\chunk_with_keys(['a' => 1, 'b' => 2, 'c' => 3], 2)
+ *     => Iter(['a' => 1, 'b' => 2], ['c' => 3])
+ *
+ * @template Tk
+ * @template Tv
+ *
+ * @param iterable<Tk, Tv> $iterable The iterable to chunk
+ * @param int $size The size of each chunk
+ *
+ * @return list<array<Tk, Tv>>
+ *
+ * @throws Psl\Exception\InvariantViolationException If $size is negative.
+ */
+function chunk_with_keys(iterable $iterable, int $size): array
```

Comment:
Should this be a vec? The result indeed is a list, but it operates on a dict input?

Reply:
we don't care about what we operate on, we care about the return type.

If return type is vec or pair of vecs, it should be in Vec\